src/TraceEvent/EventPipe/EventPipeEventSource.cs

```diff
@@ -424,7 +424,10 @@ public EventPipeEventMetaData(PinnedStreamReader reader, int length, int fileFor
         public int MetaDataId { get; private set; }
         public string ProviderName { get; private set; }
         public string EventName { get; private set; }
+#if SUPPORT_V1_V2
```

Comment:
You are leaving behind the NewTemplate() code in EventPipeTraceEventParser (which is why you need these ParameterDefintions), but basically cloning it below in ParseFields.   Given that we will be supporting V1_V2 for a while I would really like to avoid the cloning of this code.   

It seems to me that there are no incompatible differences between V3 and V2 as far as reading fields and thus making a Template, so we should be able to do the same logic for all.    Now maybe I am missing something, but I would at least like to make the attempt.